### ISSUE TITLE
Unicode strings are not cut cleanly in main view "Last post" section

### ISSUE THREAD

**Issue by rturja** [SimpleMachines/SMF#3669]
See the picture; o with umlauts "ö" gives unexpected result. I've fixed this in 2.0.x, but can't remember where I did fix it...

![unicode](https://cloud.githubusercontent.com/assets/23266104/20042480/44418fb2-a483-11e6-8bf6-172048c58881.png)

**Comment by albertlast** [comment]
Whiche collate/charset got your tabels?

**Comment by rturja** [comment]
fi_FI.UTF-8

**Comment by albertlast** [comment]
Than should be this no db related issue

**Comment by rturja** [comment]
Yep, this is in code and I have fixed this in 2.0.x since we went from ISO to UTF. Just can't find the spot and fix at the moment (and although the bug seems minor, it does annoy the users a lot...)

**Comment by rturja** [comment]
Looks like global_character_set has never found it's way to the smf_settings table. Fixed after creating and setting the variable to UTF-8.

Could have something to do with running native UTF-8 postgres. (I inherited the board after which seemed like semi-failed UTF-8 conversion in MySQL - later migrated to Postgres for less administration fuss.)

**Comment by albertlast** [comment]
is ther somehting we could fix or can we close this issue?

**Comment by rturja** [comment]
I have a faint recollection that convert to UTF-8 is working (or was on MySQL only. Also if user sets up the database in UTF (never converting it) and then goes straight to UTF version of language. In either of these cases the global_character_set setting might never be set, causing the issue shown in the image in original post. I can't be sure if I'm wrong on these use cases - can test them tomorrow though if needed.

**Comment by rturja** [comment]
Okay - I did remember right. I created blank database in PgSQL with UTF-8 and Finnish collation. Then installed 2.0.12 full, selecting UTF-8 in install. After installation global_character_set isn't even created in smf_settings and the forum doesn't give option to change forum to UTF-8 in maintenance. (Supposedly SMF can't decide if UTF-8 is or isn't supported and errs on caution.)

Then I updated the forum to 2.1. global_character_set isn't created in the settings table.

Postgres does pretty much throw a fit if wrong character set is stored into base, so when the collation and database character set is set to UTF and also UTF language has been set the forum behaves as expected, except for the string handling - for some reason 

`$utf8 = (empty($modSettings['global_character_set']) ? $txt['lang_character_set'] : $modSettings['global_character_set']) === 'UTF-8';`

from Sources/Load.php evaluates that we're not in multibyte environment, even if lang_character_set equal UTF-8 thus making the string handlers behave weirdly.

//EDIT: If I install 2.1 from scratch - exactly same parameters in DB setup etc. and enable UTF-8, the global character set is initialised properly as expected.
